core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -251,4 +251,18 @@ class ChannelSuite extends Fs2Suite {
 
     TestControl.executeEmbed(test)
   }
+
+  test("sendAndClose closes right after sending the last element") {
```

Comment:
Maybe I missed it but I don't think this test exercises the race condition that rules out an implementation like `send(...) *> close`.

I was thinking we can write a test like:
```scala
ch.send(0).both(ch.sendAndClose(1)).product(ch.stream.compile.toList)
```
The result of that should be one of:
```scala
((Right(()),Right(())),List(0, 1))

((Left(Closed),Right(())),List(1))
```

Any other result should be impossible I believe. The trivial `send(...) *> close` would fail that test.